src/main/scala/org/wikimedia/analytics/hdfstools/HdfsRsyncExec.scala

```diff
@@ -378,49 +350,141 @@ class HdfsRsyncExec(config: HdfsRsyncConfig) {
         }
     }
 
+    /**
+     * Function listing files and folders from a list of sources.
+     * It maintains for every file its basePath (root of the transfer)
+     * to possibly use it in filter-rules.
+     * The found files and folders are orgnised in a map keyed by filename, containing
+     * Seq(filestatus, basePath) as values. We need sequences as values to correctly
+     * handing merging sources folder when they have the same name.
+     *
+     * @param srcPathList the list of sources to scan
```

Comment:
It took me a minute to realize that the the second item in the tuple was the base path.   Document that here?